device/config/configurator.go

```diff
@@ -1,93 +1,86 @@
 package config
 
 import (
+	"fmt"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/pkg/errors"
 	"github.com/xshellinc/tools/dialogs"
+	"github.com/xshellinc/tools/lib/help"
+	"github.com/xshellinc/tools/lib/ping"
 	"github.com/xshellinc/tools/lib/ssh_helper"
+	"github.com/xshellinc/tools/locale"
+	"sort"
 )
 
-// Default Configurator constants that are describe a specific configuration option
-const (
-	Locale = iota
-	Keymap
-	Wifi
-	Interface
-	DNS
-)
-
-// consts are a const strings representation
-var consts = [...]string{
-	"Locale",
-	"Keymap",
-	"Wifi",
-	"Interface",
-	"DNS",
-}
-
-// GetConstLiteral gets a literal from configurator.const
-func GetConstLiteral(v int) string {
-	return consts[v]
-}
-
 type (
-	// Configurator is a config helper, which uses CallbackFn to store configurations for devices
-	Configurator interface {
-		Setup() error
-		Write() error
-		SetConfigFn(int, *CallbackFn)
-		GetConfigFn(int) *CallbackFn
-		AddConfigFn(*CallbackFn)
-		RemoveConfigFn(int)
-	}
-
 	// configurator is a container of a mutual storage and order of CallbackFn
 	configurator struct {
 		storage map[string]interface{}
-		order   []*CallbackFn
+		order   map[string]*callbackFn
 	}
 
-	// Function is a function with an input parameter of configurator's `storage`
-	Function func(map[string]interface{}) error
+	// cb is a function with an input parameter of configurator's `storage`
+	cb func(map[string]interface{}) error
 
 	// CallbackFn is an entity with Config and Apply function
-	CallbackFn struct {
-		Config Function
-		Apply  Function
+	callbackFn struct {
+		Config cb
+		Apply  cb
 	}
-)
 
-// NewDefault creates a default Configurator
-func NewDefault(ssh ssh_helper.Util) Configurator {
+	// Interfaces represents network interfaces used to setup devices
+	Interfaces struct {
+		Address string
+		Gateway string
+		Netmask string
+		DNS     string
+	}
+)
 
-	s := make(map[string]interface{})
+// New creates an empty Configurator
+func New(ssh ssh_helper.Util) *configurator {
```

Comment:
__[golint]__
* [golint] exported func New returns unexported type *config.configurator, which can be annoying to use

<!-- Comment by [SideCI](https://sideci.com) -->